### ISSUE TITLE
[Bug]: Video without valid channel page fails to download

### ISSUE THREAD

**Issue by x775** [tubearchivist/tubearchivist#320]
### Latest and Greatest

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).

### Operating System

Ubuntu 22.04 LTS, but running using `docker-compose`.

## Describe the bug

This is, admittedly, a weird one and definitely an outlier. However, `KeyError: 'title'` is thrown attempting to download.

## Steps To Reproduce

I can consistently reproduce this error with https://www.youtube.com/watch?v=8kfrTGKz6Ww. 

That said, I am not actually sure what is happening with this particular YouTube video. Again, this is a special case.

## Expected behavior

Since the video is indexed and correctly shows in TA, it is reasonable to assume that it is also downloadable. 

![image](https://user-images.githubusercontent.com/9308240/186271622-43e556b7-58ef-440f-887b-b6702418cad7.png)

### Relevant log output

```shell
tubearchivist      | [2022-08-23 23:36:21,708: ERROR/ForkPoolWorker-1] Task home.tasks.download_single[4f7812f8-8367-4771-8078-3615b6395463] raised unexpected: KeyError('title')
tubearchivist      | Traceback (most recent call last):
tubearchivist      |   File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
tubearchivist      |     R = retval = fun(*args, **kwargs)
tubearchivist      |   File "/root/.local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
tubearchivist      |     return self.run(*args, **kwargs)
tubearchivist      |   File "/app/home/tasks.py", line 119, in download_single
tubearchivist      |     VideoDownloader().run_queue()
tubearchivist      |   File "/app/home/src/download/yt_dlp_handler.py", line 178, in run_queue
tubearchivist      |     vid_dict = index_new_video(
tubearchivist      |   File "/app/home/src/index/video.py", line 388, in index_new_video
tubearchivist      |     video.build_json()
tubearchivist      |   File "/app/home/src/index/video.py", line 141, in build_json
tubearchivist      |     self._add_channel()
tubearchivist      |   File "/app/home/src/index/video.py", line 195, in _add_channel
tubearchivist      |     channel.build_json(upload=True, fallback=self.youtube_meta)
tubearchivist      |   File "/app/home/src/index/channel.py", line 182, in build_json
tubearchivist      |     self.get_from_youtube(fallback)
tubearchivist      |   File "/app/home/src/index/channel.py", line 190, in get_from_youtube
tubearchivist      |     self.json_data = ChannelScraper(self.youtube_id).get_json()
tubearchivist      |   File "/app/home/src/index/channel.py", line 43, in get_json
tubearchivist      |     self._parse_channel_main()
tubearchivist      |   File "/app/home/src/index/channel.py", line 92, in _parse_channel_main
tubearchivist      |     "channel_name": main_tab["title"],
tubearchivist      | KeyError: 'title'
```

**Comment by bbilly1** [comment]
Thanks for reporting this. I can reproduce that as well. I think the channel page can't get parsed correctly for some reason. We'll need to look into it.

**Comment by bbilly1** [comment]
Did some debugging into this one. Looks like that video belongs to a channel, that doesn't exist, but the video is there. yt-dlp detects the channel URL https://www.youtube.com/channel/UCNpZdhOqTQqaE1D_WjdZ46g and uploader URL http://www.youtube.com/user/AgeOfChanceVEVO, but both of these links return *This channel is not available*.

Related to a different issue #312, I've added some better error handling for channel metadata scraping: https://github.com/tubearchivist/tubearchivist/commit/a6dd9278600aad57ec0b629c2d352b8fa09e87e2.

This will allow you to successfully download, index and play that video and Tube Archivist will create a channel page with as much metadata as there is available and will fallback to the default channel art. 

So to summarize, thank you for raising the issue, but as luck would have it, that error is already handled in a fix for an unrelated error.

**Comment by bbilly1** [comment]
The fix for that error is now merged in v0.2.2, thanks for reporting, please update.
